control-announcements/src/main/java/org/oskari/announcements/actions/AnnouncementsHandler.java

```diff
@@ -0,0 +1,101 @@
+package org.oskari.announcements.actions;
+
+import fi.nls.oskari.control.RestActionHandler;
+import org.oskari.announcements.helpers.AnnouncementsParser;
+import org.oskari.announcements.helpers.Announcement;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.ActionException;
+import fi.nls.oskari.control.ActionParameters;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.oskari.log.AuditLog;
+import fi.nls.oskari.util.JSONHelper;
+
+@OskariActionRoute("Announcements")
+public class AnnouncementsHandler extends RestActionHandler {
+    private AnnouncementsService service = new AnnouncementsServiceMybatisImpl();
+
+    // Handle get announcements
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
+
+        try {
+            JSONObject result = new JSONObject();
+            if (params.getUser().isAdmin()) {
```

Comment:
```suggestion
            if (params.getHttpParam("all", false) && params.getUser().isAdmin()) {
```

Then on the frontend when you want to list all of the announcements for admin you can call `/action?action_route=Announcements&all=true`